src/core/Akka/Util/Internal/Extensions.cs

```diff
@@ -144,7 +144,7 @@ public static string BetweenDoubleQuotes(this string self)
         /// <returns>TBD</returns>
         public static T GetOrElse<T>(this T obj, T elseValue)
         {
-            if (obj.Equals(default(T)))
+            if (obj == null || obj.Equals(default(T)))
```

Comment:
what kind of strange behavior you have on mind? With structs the first condition will never match since they cannot be null, so this will only change behavior for classes

Reply:
Given `obj` = `0`:

    if (obj == null || obj.Equals(default(T)))
    if (0 == null || 0.Equals(default(int)))
    if (false || 0.Equals(0))
    if (false || true)
    if (true)

What I mean here is that this method has no sense in context of value types, unlike null 0 is not an absence of value. This method was ported straightforward from scala for null checks way back in the past (before elvis operator existed AFAIK).

Reply:
That would be great! 👍 Thanks for the patience.